### DISCUSSION TITLE
Default theme for Webspaces?

### DISCUSSION THREAD

**Discussion by manuxi** [sulu/sulu#7628] [General]
I have build up an installation with two webspaces. In addition I use the themes bundle to have a theme for each webspace. Basically, both webspaces with their themes run very well. But there are a few hitches in some places. 
When I'm in admin in one webspace (I'll call it webspace B) and do some translation of images from within copied pages, after a while the theme switches to the one of webspace A. Unfortunately I created webspace A first (it's a pretty small space) then the other one. Is there a possibility to set a default theme?

**Comment by alexander-schranz** [comment]
The SuluThemeBundle does nothing special as setting the `SyliusThemeBundle` theme based on the matched webspace here: https://github.com/sulu/SuluThemeBundle/blob/5f542f18df5b550af0cf159f381749b52731a35e/EventListener/SetThemeEventListener.php#L60. It is really really small bundle without lot of own functionality. So a theme match is based on your webspace URL configuration, you normally should have no conflicts between the URL configuration of your 2 webspaces else you may run into other problems with unique URLs.

How themes work is basically more documented on the Sylius side. But normally they fallback if I remember correctly to the default `templates` directory to load the twig file from there if it was not found in the theme itself. Still you can make theme inheritance possible, that is document here: https://github.com/Sylius/SyliusThemeBundle/blob/2.4/docs/theme_inheritance.md. Keep in mind the theme bundle mostly only required if you reuse the XML Template but have a whole different HTML structure. For including different CSS files I personally prefer a if else inside my base.html.twig file instead of overhead of managing themes.

**Comment by manuxi** [comment]
Thank you for your answer. 
[I went through the Sylius documentation some weeks ago and am happy to work with the themes since they work very well. I let Weebpack do the work with some minor changes in CSS - overwriting some of the CSS from the default theme. For the templates I extracted some parts in the default to have less to overwrite. This works in 99% of the cases (Some templates aren't fetched correctly in the preview).]
I faced the problem only in the admin interface when doing content manager stuff: after a while the theme in the preview changed to theme A - although im on www.B.com/admin.

**Reply by alexander-schranz** [reply]
The Preview is handled in the same listener some lines at the bottom you may want to debug it here to check when it switches: https://github.com/sulu/SuluThemeBundle/blob/5f542f18df5b550af0cf159f381749b52731a35e/EventListener/SetThemeEventListener.php#L69.
